src/std/misc/list.ss

```diff
@@ -0,0 +1,92 @@
+;; -*- Gerbil -*-
+package: std/misc
+;;;; List utilities
+
+(export
+  plist->alist
+  length=? length=n?
+  length<? length<n? length<=? length<=n?
+  length>? length>n? length>=? length>=n?
+  call-with-list-builder
+  snoc append1)
+
+(import
+  :std/error :std/srfi/1)
+
+;; This function transform a property list (k1 v1 k2 v2 ...) into
+;; an association list ((k1 . v1) (k2 . v2) ...).
+;; NB: it uses equal? as the equality predicate for keys.
+(def (plist->alist plist)
+  (let loop ((p plist))
+    (match p
+      ([k v . rest] (cons (cons k v) (loop rest)))
+      ([] [])
+      ([_] (error "Odd number of elements in plist" plist)))))
+
+
+;; Are the two lists of the same length. Note: diverges if either list is circular.
+(def (length=? x y) ;; Same as (= (length x) (length y))
+  (let ((nx? (null? x))
+        (ny? (null? y)))
+    (cond
+     (nx? ny?)
+     (ny? #f)
+     (else (length=? (cdr x) (cdr y))))))
+
+;; Is the list of a given length?
+(def (length=n? x n) ;; (= (length x) n)
+  (cond
```

Comment:
i think we can use fx primitives for the arithmetic operations for slightly more efficient operation.

Reply:
Done.